android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java

```diff
@@ -30,10 +31,12 @@
     private final ConnectivityNetworkCallback mNetworkCallback;
     Network mNetwork = null;
     NetworkCapabilities mNetworkCapabilities = null;
+    public ConnectivityManager connection;
 
     public NetworkCallbackConnectivityReceiver(ReactApplicationContext reactContext) {
         super(reactContext);
         mNetworkCallback = new ConnectivityNetworkCallback();
+        connection = (ConnectivityManager) reactContext.getSystemService(Context.CONNECTIVITY_SERVICE);
```

Comment:
There's no need to create this class member here as it's already in the super class (`ConnectivityReceiver`). You can just use `getConnectivityManager()` in your new code below.

Reply:
Yeah alright, thank you for having spent time to review my modification and for suggesting me about `getConnectivityManager()`, I will definetly use this method. Anyway, I was in this scenario where I needed to programmatically switch to a know WiFi AP (without Internet access), which is an IoT thing I need to send a configuration to through a post request, and then subsequently disconnecting from it, waiting for the phone to automatically reconnect to the old WiFi network. So I was using netinfo to check when Internet connection was back on, and when it was I needed to perform some more actions. Without this modification, whenever the phone automatically attached to the old WiFi network it was not able to send requests over the network anymore, not even to connect (for instance) to an MQTT broker (which was also my case). So I found this discussion https://stackoverflow.com/questions/65558279/machine-is-not-on-the-network-when-trying-to-connect-mqtt in which they refer to Android native code and so I tried to add it to your library and now it seems to work well. I don't know if my scenario is clear, if not let me know and I will give you more details :)